android/src/main/java/com/bitmovin/player/reactnative/BufferModule.kt

```diff
@@ -0,0 +1,73 @@
+package com.bitmovin.player.reactnative
+
+import com.bitmovin.player.api.buffer.BufferLevel
+import com.bitmovin.player.api.media.MediaType
+import com.bitmovin.player.reactnative.converter.JsonConverter
+import com.facebook.react.bridge.*
+import com.facebook.react.module.annotations.ReactModule
+import com.facebook.react.uimanager.UIManagerModule
+
+private const val MODULE_NAME = "BufferModule"
+
+@ReactModule(name = MODULE_NAME)
+class BufferModule(private val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+    /**
+     * JS exported module name.
+     */
+    override fun getName() = MODULE_NAME
+
+    /**
+     * Gets the [buffer level][BufferLevel] from the Player
+     * @param nativeId Target player id.
+     * @param type The [type of buffer][JsonConverter.toBufferType] to return the level for.
+     * @param promise JS promise object.
+     */
+    @ReactMethod
+    fun getLevel(nativeId: NativeId, type: Int, promise: Promise) {
+        uiManager()?.addUIBlock { _ ->
+            val player = playerModule()?.getPlayer(nativeId) ?: return@addUIBlock
+            val bufferType = JsonConverter.toBufferType(type)
+            val bufferLevels = RNBufferLevels(
+                player.buffer.getLevel(bufferType, MediaType.Audio),
+                player.buffer.getLevel(bufferType, MediaType.Video),
+            )
+            JsonConverter.fromRNBufferLevels(bufferLevels)?.let {
+                promise.resolve(it)
+            }
+        }
+    }
+
+    /**
+     * Sets the target buffer level for the chosen buffer type across all media types.
+     * @param nativeId Target player id.
+     * @param type The type of the buffer to set the target level for.
+     * @param value The value to set.
+     */
+    @ReactMethod
+    fun setTargetLevel(nativeId: NativeId, type: Int, value: Double) {
+        uiManager()?.addUIBlock { _ ->
+            val player = playerModule()?.getPlayer(nativeId) ?: return@addUIBlock
+            val bufferType = JsonConverter.toBufferType(type)
+            player.buffer.setTargetLevel(bufferType, value)
+        }
+    }
+
+    /**
+     * Helper function that gets the instantiated `UIManagerModule` from modules registry.
+     */
+    private fun uiManager(): UIManagerModule? =
+        context.getNativeModule(UIManagerModule::class.java)
+
+    /**
+     * Helper function that gets the instantiated `PlayerModule` from modules registry.
+     */
+    private fun playerModule(): PlayerModule? =
+        context.getNativeModule(PlayerModule::class.java)
+
+    /**
+     * Collection of [BufferLevel] objects
+     * @param audio [BufferLevel] for [MediaType.Audio].
+     * @param video [BufferLevel] for [MediaType.Video].
+     */
```

Comment:
I don't think this documentation brings a lot of value, so I would rather avoid it.
```suggestion
```
We could probably also remove the whole `RNBufferLevels` data class and just change the fromRNBufferLevels signature to accept audio and video buffer. But no string opinion.

Reply:
Address in b54a7ac9e3c764a7e419c3e11a20b683f976df96 for Android

Reply:
Upon further discussion with @rolandkakonyi, we came up with the idea of doing the same thing we do for `RNPlayerViewConfigWrapper`: explaining why this wrapper is needed.

Proposed changes in a9af0b6c5702c698be831d89811f9e20fdd2177a
Let me know your thoughts about this change